apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Query/MessageArchiveQuery.cs

```diff
@@ -37,26 +40,36 @@ public async Task<Connection<ArchivedMessageResultV2>> GetArchivedMessagesAsync(
         int? last,
         string? before,
         ArchivedMessageSortInput? order,
-        [Service] IEdiB2CWebAppClient_V2 client)
+        [Service] IMarketParticipantClient_V1 markPartClient,
+        [Service] IEdiB2CWebAppClient_V3 ediClient)
     {
+        async Task<ActorDto?> GetActorAsync(Guid? id)
+        {
+            return id.HasValue
+                ? await markPartClient.ActorGetAsync(id.Value)
+                : null;
+        }
+
+        var sender = GetActorAsync(senderId);
+        var receiver = GetActorAsync(receiverId);
         var searchCriteria = !string.IsNullOrWhiteSpace(filter)
```

Comment:
So the input "filter" is just the messageId of the wanted message? Of so. Can we give it a more precise name?

Reply:
Good point, but unfortunately not. It has to be named `filter` due to the way the table is implemented in the frontend. When you use the search field in the table view, it automatically updates the query to filter by that value - and thus it expects the query to have a parameter named `filter`. This is of course just a convenience thing so that you do not have to set up that binding for every table.

We have an ongoing discussion in the frontend guild regarding these search fields, whether they should only be free text filters or something else. So it may change in the future.